### PR TITLE
Added Cloud Metadata Scanner

### PR DESCRIPTION
**Pull request by psiinon**
Thanks to @thc202 for the magic which preserves the Host header.
I've tested this and the host header is correctly sent.
However I havnt actually got a failing server to test this on :/
Confidence is currently LOW as it just checks for a non failing response code. Hopefully we can improve that if and when we get some failing servers to test with.

### CODE REVIEW

**Comment by kingthorin** [comment]
It is only Alpha so I guess not technically required per project guidelines. But if you have a min or two to knock together a help entry that'd be handy :+1:

**Comment by psiinon** [comment]
Yeah, forgot the help :( Will add that, and the missing timing calls @thc202 pointed out...

File: src/org/zaproxy/zap/extension/ascanrulesAlpha/CloudMetadataScanner.java [hunk 1]
```diff
@@ -0,0 +1,211 @@
...
+    /**
+     * Prefix for internationalised messages used by this rule
+     */
+    private static final String MESSAGE_PREFIX = "ascanalpha.cloudmetadata.";
+    
+    private static final int PLUGIN_ID = 90034;
+    private static final String METADATA_PATH = "/latest/meta-data/";
+    private static final String METADATA_HOST = "169.154.169.254";
+
+    private static final Logger log = Logger.getLogger(CloudMetadataScanner.class);
```
[the remark this answers is not in the document]
Comment (psiinon) [reply src/org/zaproxy/zap/extension/ascanrulesAlpha/CloudMetadataScanner.java]: done

**Comment by psiinon** [comment]
Help and timing code added.

**Comment by thc202** [comment]
Thanks!

**Pull request merged**